test/dialogs/guided_setup_test.rb

```diff
@@ -25,54 +25,82 @@
 
 describe Y2Storage::Dialogs::GuidedSetup do
 
-  def run_dialog(dialog, &block)
+  def allow_run_dialog(dialog, &block)
     allow_any_instance_of(dialog).to receive(:run), &block
   end
 
-  def run_select_disks(&block)
-    run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectDisks, &block)
+  def allow_run_select_disks(&block)
+    allow_run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectDisks, &block)
   end
 
-  def run_select_root_disk(&block)
-    run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectRootDisk, &block)
+  def allow_run_select_root_disk(&block)
+    allow_run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectRootDisk, &block)
   end
 
-  def run_select_scheme(&block)
-    run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectScheme, &block)
+  def allow_run_select_scheme(&block)
+    allow_run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectScheme, &block)
   end
 
-  def run_select_filesystem(&block)
-    run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectFilesystem, &block)
+  def allow_run_select_filesystem(&block)
+    allow_run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectFilesystem, &block)
   end
 
-  subject { described_class.new(fake_devicegraph, settings) }
+  def expect_run_dialog(dialog)
+    expect_any_instance_of(dialog).to receive(:run).once
+  end
+
+  def expect_not_run_dialog(dialog)
+    expect_any_instance_of(dialog).not_to receive(:run)
+  end
+
+  def expect_not_run_select_disks
+    expect_not_run_dialog(Y2Storage::Dialogs::GuidedSetup::SelectDisks)
+  end
```

Comment:
Right now only once, but maybe in next iterations we will skip more steps of the wizard. In these cases, we will expect to skip the same dialog several times.